shell/platform/darwin/macos/framework/Source/FlutterThreadSynchronizer.h

```diff
@@ -38,6 +38,13 @@
                         size:(CGSize)size
                       notify:(nonnull dispatch_block_t)notify;
 
+/**
+ * Schedules the given block to be performed on the platform thread.
+ * The block will be performed even if the platform thread is blocked waiting
+ * for a commit.
+ */
+- (void)performOnMainThread:(nonnull dispatch_block_t)block;
```

Comment:
I'd suggest naming this `performOnPlatformThread` to avoid ambiguity in thread naming within the embedder and then in the comment say something like "the platform thread/macOS main thread" or similar.